ToggleAppliances/Main.cs

```diff
@@ -1,24 +1,28 @@
-﻿using Harmony;
-using System;
-using System.IO;
-using System.Reflection;
-
-namespace ToggleMachines
-{
-    public class Main
-    {
-        public static void Patch()
-        {
-            var harmony = HarmonyInstance.Create("com.ahk1221.toggleappliances");
-            harmony.PatchAll(Assembly.GetExecutingAssembly());
-
-            Logger.Log("Successfully patched!");
-        }
-
-        public static string GetSavePathDir()
-        {
-            var savePathDir = Path.Combine(@".\SNAppData\SavedGames\", Utils.GetSavegameDir());
-            return Path.Combine(savePathDir, "ToggleAppliances");
-        }
-    }
-}
+﻿using Harmony;
+using System;
+using System.IO;
+using System.Reflection;
+using UnityEngine;
+using QModManager.API.ModLoading;
+using SMLHelper.V2.Utility;
+namespace ToggleMachines
+{
+    [QModCore]
+    public class Qpatch
+    {
+        [QModPatch]
+        public static void Patch()
+        {
+            var harmony = HarmonyInstance.Create("com.ahk1221.toggleappliances");
+            harmony.PatchAll(Assembly.GetExecutingAssembly());
+
+            Logger.Log("Successfully patched!");
+        }
+
+        public static string GetSavePathDir()
+        {
+            var savePathDir = Path.Combine(@".\SNAppData\SavedGames\", SaveUtils.GetCurrentSaveDataDir());

```

Comment:
If you want to avoid introducing a dependency to SMLHelper on a mod that didn't require it, you can just use the underlying code of [SaveUtils.GetCurrentSaveDataDir](https://github.com/SubnauticaModding/SMLHelper/blob/master/SMLHelper/Utility/SaveUtils.cs)
```CS
SaveLoadManager.GetTemporarySavePath()
```